packages/block-library/src/heading/editor.scss

```diff
@@ -1,20 +1,16 @@
 // Remove padding in heading level control popover since the toolbar buttons already have padding.
 .block-library-heading-level-dropdown .components-popover__content {
-	// TODO: Find a less hardcoded way of doing this. `max-content` works on
-	// Chromium, but it results in a scrollbar on Safari, and isn't supported
-	// at all in IE11.
-	min-width: 230px;
-
 	> div {
 		padding: 0;
 	}
 }
 
-// The dropdown already has a border, so we can remove the one on the heading
-// level toolbar.
-.block-library-heading-level-toolbar {
-	border: none;
-	.components-toolbar-group {
-		flex-wrap: nowrap;
+.block-library-heading-level-menu-popover {
+	.components-dropdown-menu__menu,
+	.components-menu-item__button {
```

Comment:
Let's move all of these styles into `DropdownMenu` when no text is supplied (only icon buttons).